### PR TITLE
Adding legacy-endpoint-access sample

### PR DESCRIPTION
**Pull request by chansmith97**
Adding a sample showing how to use the new legacy-endpoint-access endpoint to detect usage of the legacy metadata endpoints.

### CODE REVIEW

File: compute/metadata/detect_legacy_usage.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import json
+import requests
+import time
+
+METADATA_URL = 'http://metadata.google.internal/computeMetadata/v1/'
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+
+
+def wait_for_legacy_usage(callback):
```
[the remark this answers is not in the document]
Comment (chansmith97) [reply compute/metadata/detect_legacy_usage.py]: Added a brief explanation and region tags. Should I also include a link to the page the sample will be included on?

File: compute/metadata/detect_legacy_usage_test.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import mock
+import requests
+import json
+
+import detect_legacy_usage
+
+@mock.patch('detect_legacy_usage.requests')
+@mock.patch('detect_legacy_usage.time')
+def test_wait_for_legacy_usage(time_mock, requests_mock):
```
[the remark this answers is not in the document]
Comment (chansmith97) [reply compute/metadata/detect_legacy_usage_test.py]: Broke into 4 test cases

File: compute/metadata/detect_legacy_usage.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    url = METADATA_URL + 'instance/legacy-endpoint-access/'
```
[the remark this answers is not in the document]
Comment (chansmith97) [reply compute/metadata/detect_legacy_usage.py]: Done.

File: compute/metadata/detect_legacy_usage.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kurtisvg) [reply compute/metadata/detect_legacy_usage.py]: Usually we do so in the README, but it's not required.

**Pull request merged**